src/Neo.Compiler.CSharp/MethodConvert/System/SystemCall.Register.cs

```
@@ -484,32 +265,56 @@ private static void RegisterNumericTypeHandlers()
-        RegisterHandler((short value) => short.PopCount(value), HandleShortPopCount);
-        RegisterHandler((ushort value) => ushort.PopCount(value), HandleUShortPopCount);
-        RegisterHandler((int value) => int.PopCount(value), HandleIntPopCount);
-        RegisterHandler((uint value) => uint.PopCount(value), HandleUIntPopCount);
-        RegisterHandler((long value) => long.PopCount(value), HandleLongPopCount);
-        RegisterHandler((ulong value) => ulong.PopCount(value), HandleULongPopCount);
         RegisterHandler((BigInteger value) => BigInteger.PopCount(value), HandleBigIntegerPopCount);
         #endregion PopCount
     }
 
+    private static void RegisterCreateCheckedHandlers(Type targetType, SystemCallHandler handler)
+    {
+        RegisterCreateNumericHandlers("CreateChecked", targetType, handler);
+    }
+
+    private static void RegisterCreateSaturatingHandlers(Type targetType, SystemCallHandler handler)
+    {
+        RegisterCreateNumericHandlers("CreateSaturating", targetType, handler);
+    }
+
+    private static void RegisterCreateNumericHandlers(string methodName, Type targetType, SystemCallHandler handler)
+    {
+        foreach (var sourceType in s_numericCreateSourceTypes)
+            RegisterCreateNumericHandler(methodName, targetType, sourceType, handler);
+    }
+
+    private static void RegisterCreateNumericHandler(string methodName, Type targetType, Type sourceType, SystemCallHandler handler)
+    {
+        var methodDefinition = targetType.GetMethods(BindingFlags.Public | BindingFlags.Static)
+            .FirstOrDefault(m => m.Name == methodName && m.IsGenericMethodDefinition && m.GetParameters().Length == 1);
+        if (methodDefinition is null)
+            return;
+
+        MethodInfo method;
+        try
+        {
+            method = methodDefinition.MakeGenericMethod(sourceType);
+        }
+        catch (ArgumentException)
+        {
+            return;
```

Silent exceptions during registration could result in a wrong script

---

This is not user contract compiling logic, but internal system method registering logic, basically has nothing to do with user script, only issue might be `no such syntax ` exception, instead of wrong scirpt .

---

I know, but it's for us, seems that ` var leadingZeroCountMethod = type.GetMethod("LeadingZeroCount", new[] { type });` should throw an exception, because always should exists in some types